modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -1758,6 +1759,19 @@ private ObjectDefinition _publishObjectDefinition(
 		_registerTransactionCallbackForCluster(
 			_deployObjectDefinitionMethodKey, objectDefinition);
 
+		if (objectDefinition.isRootDescendantNode()) {
```

Comment:
Why do we need to create and delete it right after?

Is there a way to avoid creating the ResourcePermissions instead of deleting them?

Reply:
I didn't find where these ResourcePermissions are created, I have a feeling that isn't by our side, so I chose to get them and delete them. But I can take a look deeper if you believe it is better